pkg/quay/quay.go

```diff
@@ -0,0 +1,166 @@
+package quay
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/giantswarm/microerror"
+)
+
+const (
+	publicImagesOnly   = true
+	registryEndpoint   = "https://quay.io"
+	repositoryEndpoint = "https://quay.io/api/v1/repository"
+
+	tagLengthLimit = 20
+)
+
+type Quay struct {
+	namespace    string
+	lastModified time.Duration
+
+	httpClient *http.Client
+}
+
+type Config struct {
+	Namespace    string
+	LastModified time.Duration
+}
+
+type Repository struct {
+	Name         string `json:"name"`
+	LastModified int    `json:"last_modified"`
+}
+
+type RepositoriesJSON struct {
+	Repositories []Repository `json:"repositories"`
+}
+
+func New(c Config) (*Quay, error) {
+	httpClient := &http.Client{}
+
+	return &Quay{
+		namespace:    c.Namespace,
+		lastModified: c.LastModified,
+
+		httpClient: httpClient,
+	}, nil
+}
+
+func (q *Quay) Authorize() error {
+	return nil
+}
+
+func (q *Quay) ListRepositories() ([]string, error) {
+	fmt.Printf("Reading list of quay repostories in %#q namespace...\n", q.namespace)
+
+	var reposToSync []string
+
+	req, err := http.NewRequest("GET", repositoryEndpoint, nil)
+	if err != nil {
+		return reposToSync, microerror.Mask(err)
+	}
+
+	query := req.URL.Query()
+	query.Add("last_modified", "true")
+	query.Add("starred", "false")
+	query.Add("public", fmt.Sprintf("%t", publicImagesOnly))
+	query.Add("namespace", q.namespace)
+	req.URL.RawQuery = query.Encode()
+
+	resp, err := q.httpClient.Do(req)
+	if err != nil {
+		return reposToSync, microerror.Mask(err)
+	}
+
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return reposToSync, microerror.Mask(err)
+	}
+
+	var data RepositoriesJSON
+
+	err = json.Unmarshal(body, &data)
+	if err != nil {
+		return reposToSync, microerror.Mask(err)
+	}
+
+	for _, repo := range data.Repositories {
+		lastModifiedTimestamp := int(time.Now().Add(-1 * q.lastModified).Unix())
+		if repo.LastModified > lastModifiedTimestamp {
+			reposToSync = append(reposToSync, fmt.Sprintf("%s/%s", q.namespace, repo.Name))
+		}
+	}
+
```

Comment:
Don't we have to do `resp.Header.Get("Link")` thing here?

Reply:
For now, it doesn't give more than one page. So I can't even test it of there would be more repos to load